src/Lachain.Core/Vault/KeyGenManager.cs

```diff
@@ -380,7 +381,7 @@ public bool RescanBlockChainForKeys(IPublicConsensusKeySet publicKeysToSearch)
                     {
                         Logger.LogDebug(
                             $"Detected call of GovernanceContract.{GovernanceInterface.MethodKeygenCommit}");
-                        var sender = keygen.GetSenderByPublicKey(tx.RecoverPublicKey());
+                        var sender = keygen.GetSenderByPublicKey(tx.RecoverPublicKey(HardforkHeights.IsHardfork_8Active(i + 1)));
```

Comment:
Should it be ' i ' ? because the block index here is ' i ' .

Reply:
fixed